src/components/pages/Expeditions/CreationDialog/ConfigImport/expeditionJsonSchema.ts

```diff
@@ -86,10 +86,7 @@ export const schema = {
     Barracks: {
       properties: {
         mageIds: {
-          items: {
-            type: 'string',
-          },
-          type: 'array',
+          $ref: '#/definitions/Array_1',
```

Comment:
Unfortunately this wont do.
There are a few more things necessary to make all of this work:

We will probably need to:
1. Add another type for our `initialBarracksConfig` so that it is no longer of the type `Barracks` but the type `InitialBarracksConfig` in here https://github.com/on3iro/aer-types/blob/master/types/expeditions.ts

The type will probably look like this:

```ts
type InitialBarracksConfig = {
   mageIds: [
       { type: 'random' },
       { type: 'custom', id: 'someMage' }
   ],
   supplyIds: string[],
   treasureIds: string[]
}
```

2. We then need to create a new version of the aer-types package and publish it to npm (I will need to do that)
3. You will need to install the new version with **yarn**
4. You might then see new typescript errors in the compiler - so you will probably need to make adjustments to your code
5. You then need to re-create the json-schema with the command inside the comment above

We need to do this, to make the app as typesafe as possible (during compilation as  well as at runtime, when users try to import an expedition)

6. We also need to change the documentation for initial barracks, so its in line with the change

(7.) It would probably make sense, to  also allow treasureIds and supplyIds to be randomly rolled in that scenario. So maybe we should change these in the `InitialBarracksConfig` as well (which in turn would mean, that you would need to provide additional code for them) - However we can focus on mages now and add that later, if you would prefer to keep this PR streamlined :)

Reply:
If you need help here, just let me know. We can also make a call over on discord, so I could guide you through necessary steps, if that would help you :)

Reply:
I think a discord call would be good here to show me some of this process. The other issues above are things I can easily amend before our call but the relationship between the aer-types package and the json-schema are something I'm still pretty fuzzy on. 

I'm thinking this weekend is the time that makes the most sense to try to get together over discord. I'm also starting to think we might be in pretty different timezones. I live in Arizona so I use MST (Mountain Standard Time). Not sure what time would maybe work for you or if you want to play it by ear.

And I don't mind trying to get supplyids and treasureids working as well.

Reply:
Cool - yeah we definitely would have to check timezones before hand :D

I am currently thinking about migrating aer-types back into this repository (but keep publishing it as a separate type-library to npm...). That would make collaborating way easier. 

I probably wont be able to make a call during the upcoming weekend because of the timezone issue (this weekend is rather full either) so maybe the weekend after? I will definitely review your code this weekend and leave a few more comments, though. So that way you should be able to already make a few more adjusments. Would that work for you? 

(unfortunately you catched me durng a rather busy time with project dead lines at work as well as lots of family stuff to attend to in my spare time)

Reply:
(we are nine hours apart btw - so when its 10:00 am in your country its 7:00pm in mine)

Reply:
@Quicksav I changed the way this code base works regarding types and data. Both are now part of the repository again (we will still publish them to NPM, so that they can be consumed by other libraries). This will make it way easier to adjust typing while working on a feature, because we no longer need to publish the typings in between etc.

For you to receive these changes, you need to do the following things on your feature branch:

1. set a git remote for this original repository (in addition to your own fork upstream remote), name it `upstream` or something like that
2. than on your feature branch run `git pull --rebase upstream dev` - that should get you all the changes you need
3. run `yarn` again, so that your _node_modules_ are up to date

Unfortunately I am out of time for now and wasn't been able to look at your changes. I will try to do so asap. But if you want you could play around with the typings a bit and try to type the initialBarracksConfig.

If you have any questions, let me know :)

Reply:
Having some trouble understanding what I'm supposed to do with this. Unfortunately I think most of my confusion stems from my unfamiliarity with proper git usage. Really just not understanding number 1 of the above steps. Am I supposed to fork the original repository again? Am i just making an 'upstream' branch on my local  machine? Or is it that I need to make the 'upstream' branch remote? Sorry for my ineptitude with this I am used to using Visual Studio to handle git commands in a relatively straightforward development environment.

Would having a call next Saturday at ~10AM your time (it'd be Friday after midnight for me) work to go over some of this? I've fixed the other feedback locally but this is the big outstanding thing.

Reply:
Hey, no problem :)

So the thing we are trying to achieve is, to get all the new changes made to this repository (so everything which has been merged to `dev`) into the feature branch you are currently working on.
To do this, you have to follow a few steps.

The first step outlined above, is that you need this repository as a remote source for fetching changes.
Currently your local repository probably only has a single remote (which is probably called `origin`). A remote is a server you can connect to and push/pull changes/branches/tags etc. from/to. This single remote targets your fork an github. When you type

`git remote -v show` into your terminal, you will see something like this:

![Screenshot 2020-10-12 at 08 16 54](https://user-images.githubusercontent.com/8681413/95711567-4f7fc000-0c63-11eb-9349-c4b3f6846bd9.png)

However the url to will point to your forked repository, not our original aeons-end-randomizer repo.

So what you will have to do, is the following (locally):

1. run `git remote add upstream git@github.com:on3iro/aeons-end-randomizer.git` (note, that if you use **https** instead of **ssh** your url will look like this: `https://github.com/on3iro/aeons-end-randomizer.git`)
2. now you have an additional remote
3. now on your feature branch do the following `git pull --rebase upstream dev`
4. this rebases all changes which where made to dev into your feature branch
5. Now you need to push all of this to your fork, to update this Pull Request. Because **rebase** changes the git history, you need to do a force push, so on your feature branch do the following: `git push -f` (**note**: rebasing is very powerful, because it helps keeping a clean git history, however there are a few gotchas, so if you start rebasing in other personal projects, make sure you know these. If you want to know more, I can teach you :) )

Afterwards the PR should be mergable to `dev` again, and not longer out of date.
Does that help? :)